elyra/pipeline/component_parameter.py

```diff
@@ -169,129 +262,132 @@ def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, e
         """
         pass
 
-    def get_all_validation_errors(self) -> List[str]:
-        """Perform custom validation on an instance."""
-        return []
-
-    def is_empty_instance(self) -> bool:
-        """Returns a boolean indicating whether this instance is considered a no-op."""
-        return False
-
 
 class DisableNodeCaching(ElyraProperty):
-    """Disable caching to force node re-execution in the target runtime environment."""
+    """An ElyraProperty representing node cache preference"""
+
+    applies_to_generic = False
+    applies_to_custom = True
 
     property_id = DISABLE_NODE_CACHING
-    generic = False
-    custom = True
-    _display_name = "Disable node caching"
+    property_display_name = "Disable node caching"
+    property_description = "Disable caching to force node re-execution in the target runtime environment."
     _json_data_type = "string"
 
     def __init__(self, selection, **kwargs):
         self.selection = selection == "True"
 
     @classmethod
-    def create_instance(cls, prop_id: str, value: Optional[Any]) -> DisableNodeCaching:
+    @overrides
+    def get_single_instance(cls, value: Optional[Any] = None) -> ElyraProperty:
         return DisableNodeCaching(selection=value)
 
     @classmethod
+    @overrides
     def get_schema(cls) -> Dict[str, Any]:
-        """Build the JSON schema for an Elyra-owned component property"""
         schema = super().get_schema()
         schema["enum"] = ["True", "False"]
         return schema
 
+    def get_value_for_display(self) -> Dict[str, Any]:
+        return self.selection
+
+    def get_all_validation_errors(self) -> List[str]:
+        return []
+
     def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, execution_object: Any, **kwargs):
-        """Add DisableNodeCaching info to the execution object for the given runtime processor"""
         runtime_processor.add_disable_node_caching(instance=self, execution_object=execution_object, **kwargs)
 
 
-class ElyraPropertyListItem(ElyraProperty):
+class ElyraPropertyListItem(ElyraProperty, ABC):
```

Comment:
nit: Because this class no longer has abstract methods, its inheritance from `ABC` is unnecessary.  Unfortunately, you can't prevent a subclass of `ABC` from being instantiated unless it also defines at least one abstract method in Python.